packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/TypeProvider.cs

```diff
@@ -99,7 +102,7 @@ private TypeSignatureModifiers GetDeclarationModifiersInternal()
             // mask & (mask - 1) gives us 0 if mask is a power of 2, it means we have exactly one flag of above when the mask is a power of 2
             if ((mask & (mask - 1)) != 0)
             {
-                throw new InvalidOperationException($"Invalid modifier {modifiers} on TypeProvider {Type.Namespace}.{Name}");
```

Comment:
Including the Type property in the exception results in an infinite loop since the CSharpType will need to be constructed using the DeclarationModifiers.